enterprise/fulltext-addon/src/main/java/org/neo4j/kernel/api/impl/fulltext/ScoreEntityIterator.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright (c) 2002-2018 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.kernel.api.impl.fulltext;
+
+import java.util.Iterator;
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.Spliterator;
+import java.util.Spliterators;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+
+import org.neo4j.helpers.collection.Pair;
+import org.neo4j.kernel.api.impl.index.collector.ValuesIterator;
+
+public class ScoreEntityIterator implements Iterator<Pair<Long,Float>>
```

Comment:
Except for high level integration test that shows how it works, in most basic scenario, there are no tests for more advanced scenario, for example, iterators combine case